common/main/cpp/native_c4.cc

```diff
@@ -98,43 +99,36 @@ void litecore::jni::jniLog(const char *fmt, ...) {
     va_end(args);
 }
 
-static bool detach(jint getEnvStat) {
-    if (getEnvStat == JNI_EDETACHED) {
-        if (gJVM->DetachCurrentThread() == 0) return true;
-        C4Warn("logCallback(): doRequestClose(): Failed to detach the current thread from a Java VM");
-    }
-    return false;
+
+//-------------------------------------------------------------------------
+// Logging Callback
+//-------------------------------------------------------------------------
+
+static bool detach(jint envState) {
+    return (envState == JNI_EDETACHED) && (JNI_OK == detachJVM("log"));
```

Comment:
From the meaning of `JNI_EDETACHED`, I understand that JVM is not currently attached. If that true, should it be that `return (envState == JNI_EDETACHED) || (JNI_OK == detachJVM("log"));` instead?

Reply:
When comparing this with the old code, look like the behavior is the same BUT it's just confusing as when I read the code I thought that `envState` is the current state but it turns out that it tells you what to do.

Reply:
See above: JNI_EDETACHED describes the state of the thread when it entered the callback, not its current state.  If it was JNI_EDETACHED, it needs to be detached again, on the way out.